apps/cowswap-frontend/src/modules/tradeWidgetAddons/containers/MetamaskTransactionWarning/index.tsx

```diff
@@ -23,13 +27,32 @@ const NetworkInfo = styled.div`
 `
 
 export function MetamaskTransactionWarning({ sellToken }: { sellToken: Currency }) {
+  const [widgetProviderMetaInfo, setWidgetProviderMetaInfo] = useState<ProviderMetaInfoPayload | null>(null)
+
+  const provider = useWalletProvider()
   const walletDetails = useWalletDetails()
   const isMetamaskBrowserExtension = useIsMetamaskBrowserExtensionWallet()
   const isMetamaskViaWalletConnect = walletDetails.walletName === METAMASK_WALLET_NAME
 
-  const isMetamask = isMetamaskBrowserExtension || isMetamaskViaWalletConnect
+  const rawProvider = provider?.provider as unknown
+  const isWidgetMetamaskBrowserExtension = widgetProviderMetaInfo?.providerEip6963Info?.rdns === METAMASK_RDNS
+  const isWidgetMetamaskViaWalletConnect = widgetProviderMetaInfo?.providerWcMetadata?.name === METAMASK_WALLET_NAME
```

Comment:
As I mentioned in the other PR, I rather do a regex matching instead of only exact matches.